lua/gen/init.lua

```diff
@@ -41,54 +67,111 @@ local function get_window_options()
     end
 
     return {
-        relative = 'cursor',
+        relative = "cursor",
         width = new_win_width,
         height = new_win_height,
         row = new_win_row,
         col = 0,
-        style = 'minimal',
-        border = 'single'
+        style = "minimal",
+        border = "single",
     }
 end
 
-M.model = 'mistral:instruct'
-M.command = 'ollama run $model $prompt'
-M.command_container = 'docker exec $container ollama run $model $prompt'
-M.container = nil
+function write_to_buffer(lines)
+    if M.result_buffer ~= nil and not vim.api.nvim_buf_is_valid(M.result_buffer) then
+        return
+    end
+
+    local all_lines = vim.api.nvim_buf_get_lines(M.result_buffer, 0, -1, false)
+
+    local last_row = #all_lines
+    local last_row_content = all_lines[last_row]
+    local last_col = string.len(last_row_content)
+
+    local text = table.concat(lines or {}, "\n")
+
+    vim.api.nvim_buf_set_option(M.result_buffer, "modifiable", true)
+    vim.api.nvim_buf_set_text(M.result_buffer, last_row - 1, last_col, last_row - 1, last_col, vim.split(text, "\n"))
+    vim.api.nvim_buf_set_option(M.result_buffer, "modifiable", false)
+end
+
+function create_window(opts, job_id)
+    if M.display_mode == "float" then
+        if M.result_buffer then
+            vim.api.nvim_buf_delete(M.result_buffer, { force = true })
+        end
+        local win_opts = vim.tbl_deep_extend("force", get_window_options(), opts.win_config)
+        M.result_buffer = vim.api.nvim_create_buf(false, true)
+        vim.api.nvim_buf_set_option(M.result_buffer, "filetype", "markdown")
+
+        M.float_win = vim.api.nvim_open_win(M.result_buffer, true, win_opts)
+    else
+        vim.cmd("vnew")
+        M.result_buffer = vim.fn.bufnr("%")
+        M.float_win = vim.fn.win_getid()
+        vim.api.nvim_buf_set_option(M.result_buffer, "filetype", "markdown")
+        vim.api.nvim_win_set_option(M.float_win, "wrap", true)
+    end
+
+    local group = vim.api.nvim_create_augroup("gen", { clear = true })
+    vim.api.nvim_create_autocmd("BufDelete", {
+        buffer = M.result_buffer,
+        group = group,
+        callback = function()
+            vim.fn.jobstop(job_id)
+            if M.float_win ~= nil and vim.api.nvim_win_is_valid(M.float_win) then
+                vim.api.nvim_win_close(M.float_win, true)
+            end
+            reset()
+        end,
+    })
+end
+
+function reset()
+    M.result_buffer = nil
+    M.float_win = nil
+    M.result_string = ""
```

Comment:
Can we put `result_string` into the closure? It doesn't need to be part of the module, no?